### PR TITLE
Add a way to unarchive timelines and delete them permanently

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/20289

Changes:
- Adds a link to `View archived timelines` in the timelines list and details modals
- Shows archived timelines with a menu to unarchive and delete them

How to test:
- Create a timeline in a collection
- Archive the timeline
- Go to `Archived timelines` modal via the header menu
- Unarchive the timeline. Make sure it's possible to undo this action.
- Delete the timeline

Screenshot 2022-03-24 at 18 03 41
Screenshot 2022-03-24 at 18 04 32
Screenshot 2022-03-24 at 18 04 38

### CODE REVIEW

File: frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.styled.tsx [hunk 1]
```diff
@@ -30,33 +30,39 @@ export const CardDescription = styled.span`
...
+    border-color: ${props => (props.to ? color("brand") : "")};
 
     ${CardIcon} {
-      color: ${color("brand")};
+      color: ${props => (props.to ? color("brand") : "")};
     }
 
     ${CardTitle} {
-      color: ${color("brand")};
+      color: ${props => (props.to ? color("brand") : "")};
```
Comment (kulyk) [inline frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.styled.tsx:65]: Looks like all the `hover` styles are needed only when `CardRoot` has a `to` prop. Should we maybe extract this block of CSS and add it conditionally? Meaning something like:

```jsx
import { css } from "@emotion/react"

const cardRootHoverStyle = css`/* all the hover styles here */`

export const CardRoot = styled(Link)`
   /* common styles */
   ${props => props.to && cardRootHoverStyle}
`
```

**Review by kulyk — APPROVED** [review]
Nice and clean as always 👍 Left a minor comment

File: frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.styled.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/timelines/collections/components/TimelineCard/TimelineCard.styled.tsx]: Agree, fixed

**Review by mazameli — COMMENTED** [review]
### Nice-to-haves: 

Would be nice if this whole element was a click target, not just the icon:

image

Same with this back link actually:

image

It'd be nice when viewing archived events for a specific timeline if the back button took you back to that timeline's event list (not to the list of timelines as it currently does):

image

**Review by mazameli — APPROVED** [review]
I noted [some nice-to-haves](https://github.com/metabase/metabase/pull/21177#pullrequestreview-922142006), but this is shippable.

**Pull request merged**